src/EStimLibrary/Core/SpatialModel/LocalizationData.cs

```diff
@@ -10,6 +10,9 @@
 public record LocalizationData(IEnumerable<int> AreasFullyContaining,
     IEnumerable<int> AreasPartiallyContaining)
 {
+    // Modify the properties using ToList() to create shallow copies during initialization
```

Comment:
What did doing this fix?

Reply:
By doing this the merge method already working with the deep copied version of the input rather than the actual input itself, just another layer of protection.